subsys/caf/modules/ble_bond.c

```diff
@@ -4,17 +4,65 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#include <bluetooth/bluetooth.h>
+
 #define MODULE ble_bond
 #include <caf/events/module_state_event.h>
+#include <caf/events/button_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_BLE_BOND_LOG_LEVEL);
+
+#define BOND_ERASE_ENABLED     IS_ENABLED(CONFIG_CAF_BLE_BOND_DEFAULT_PEER_ERASE)
+#if BOND_ERASE_ENABLED
+	#define BOND_ERASE_KEY_ID      CONFIG_CAF_BLE_BOND_DEFAULT_PEER_ERASE_KEY_ID
+	#define BOND_ERASE_MAX_UPTIME  CONFIG_CAF_BLE_BOND_DEFAULT_PEER_ERASE_TIMEOUT
+#else
+	#define BOND_ERASE_KEY_ID      0xffff
+	#define BOND_ERASE_MAX_UPTIME  0
+#endif /* BOND_ERASE_ENABLED */
+
+static bool initialized;
+static bool bond_erase_before_init;
+
+
+static void bond_erase(void)
+{
+	int err = bt_unpair(BT_ID_DEFAULT, NULL);
 
+	if (err) {
+		LOG_ERR("Cannot unpair for default ID");
+		module_set_state(MODULE_STATE_ERROR);
+	} else {
+		LOG_INF("Removed bond from default identity");
+	}
+}
 
 static bool event_handler(const struct event_header *eh)
 {
 	if (is_module_state_event(eh)) {
 		const struct module_state_event *event = cast_module_state_event(eh);
 
 		if (check_state(event, MODULE_ID(settings_loader), MODULE_STATE_READY)) {
+			initialized = true;
 			module_set_state(MODULE_STATE_READY);
+
+			if (BOND_ERASE_ENABLED && bond_erase_before_init) {
+				bond_erase();
+				bond_erase_before_init = false;
+			}
+		}
+	}
+
+	if (BOND_ERASE_ENABLED && is_button_event(eh)) {
+		const struct button_event *event = cast_button_event(eh);
+
+		if (event->pressed && (k_uptime_get() < BOND_ERASE_MAX_UPTIME)) {
```

Comment:
I don't understand. Where do you check if correct button was pressed?

Reply:
Actually nowhere. I guess I accidentally removed the check at some point. Thanks !

Reply:
Thingy has one button, so lack of this check had no effect there